packages/devtools_app/lib/src/profiler/cpu_profile_model.dart

```diff
@@ -434,6 +457,22 @@ class CpuStackFrame extends TreeNode<CpuStackFrame>
 
   final CpuProfileMetaData profileMetaData;
 
+  bool get isNative =>
+      _isNative ??= id != CpuProfileData.rootId && processedUrl.isEmpty;
+
+  bool _isNative;
+
+  bool get isDartCore => _isDartCore ??= processedUrl.startsWith('dart:');
+
+  bool _isDartCore;
+
+  bool get isFlutterCore =>
+      _isFlutterCore ??= processedUrl.startsWith('package:flutter/') ||
+          name.startsWith('flutter::') ||
```

Comment:
nit: maybe make these constants so you have an easy place to document that 
`flutter::` is the Flutter namespace in C++?